extensions/VisualEditor/ApiVisualEditor.php

```diff
@@ -367,7 +385,12 @@ public function execute() {
 		wfDebugLog( 'visualeditor', "called on '$page' with paction: '{$params['paction']}'" );
 		switch ( $params['paction'] ) {
 			case 'parse':
-				$parsed = $this->getHTML( $page, $parserParams );
+				if ( isset( $params['preload'] )) {
+					$parsed = $this->getPreloadHTML( $params );
```

Comment:
I moved the extraction of the title into the call of getPreloadHTML like suggested.

The error you mention will happen if the preload parameter is pointing to a page that doesn't exist.
I tested it (see error section here: http://visualeditor.pgroland.wikia-dev.com/wiki/Test1) and it doesn't end up with a user-facing error, it just shows empty contents in VE.

I could rewrite the code like so:

				if ( isset( $params['preload'] )) {
					$parsed = $this->getPreloadHTML( $urldecode( $params['preload'] ) );
				}

				if ( !$parsed ) 
					$parsed = $this->getHTML( $page, $parserParams );
				}

Which might be safer, but I don't see any difference in the user experience. Is there something else going on (logging?) that means that approach is better?